web/Services/LandlordService.cs

```diff
@@ -75,30 +83,71 @@ await using (var transaction = await _dbContext.Database.BeginTransactionAsync(I
 
         user.LandlordId = dbModel.Id;
 
-        return ILandlordService.LandlordRegistrationResult.Success;
+        return (ILandlordService.LandlordRegistrationResult.Success, dbModel);
     }
     
     public Task<LandlordDbModel?> GetLandlordIfExistsFromId(int id)
     {
         return _dbContext.Landlords.SingleOrDefaultAsync(l => l.Id == id);
     }
 
-    public async Task ApproveLandlord(int landlordId,  BricksAndHeartsUser user)
+    public async Task ApproveLandlord(int landlordId, BricksAndHeartsUser user)
     {
         var landlord = await GetLandlordIfExistsFromId(landlordId);
         if (landlord is null)
         {
             throw new Exception("Landlord does not exist");
         }
+
         if (landlord.CharterApproved)
         {
             throw new Exception("Landlord already approved");
         }
+
         landlord.CharterApproved = true;
         landlord.ApprovalTime = DateTime.Now;
         landlord.ApprovalAdminId = user.Id;
         await _dbContext.SaveChangesAsync();
     }
+
+    public async Task<(ILandlordService.LandlordRegistrationResult result, LandlordDbModel? landlord)> RegisterLandlord(
+        LandlordProfileModel createModel)
+    {
+        var dbModel = new LandlordDbModel
+        {
+            Title = createModel.Title,
+            FirstName = createModel.FirstName,
+            LastName = createModel.LastName,
+            CompanyName = createModel.CompanyName,
+            Email = createModel.Email,
+            Phone = createModel.Phone,
+            LandlordStatus = createModel.LandlordStatus,
+            LandlordProvidedCharterStatus = createModel.LandlordProvidedCharterStatus
+        };
+        await using (var transaction = await _dbContext.Database.BeginTransactionAsync(IsolationLevel.Serializable))
+        {
+            // Check there isn't already a Landlord with that email. Nothing depends on this currently, but it would probably mean the landlord is a duplicate
+            // This requires Serializable isolation, otherwise it will not lock any rows, and two racing registrations could create duplicate records
+            if (await _dbContext.Landlords.AnyAsync(l => l.Email == createModel.Email))
+            {
+                return (ILandlordService.LandlordRegistrationResult.ErrorLandlordEmailAlreadyRegistered, dbModel);

```

Comment:
nitpick, but don't need to return the result

```suggestion
                return (ILandlordService.LandlordRegistrationResult.ErrorLandlordEmailAlreadyRegistered, null);
```